java/arcs/core/data/expression/Evaluator.kt

```diff
@@ -92,6 +92,10 @@ class ExpressionEvaluator(
         val arguments = expr.arguments.map { it.accept(this) }.toList()
         return expr.function.invoke(this, arguments)
     }
+
+    override fun <T> visit(expr: Expression.EvalExpression<T>): Any {
+        return PaxelParser.parse(expr.paxelExpression).accept(this)
```

Comment:
A catch may be: Don't put EvalExpression in loops (FROM) cause we will re-parse it multiple times.